package message;

message Message {
    repeated int32 id = 1;
}

message SearchRequest {

    required string query = 1;

    optional int32 page_number = 2;

    optional int32 result_per_page = 3;
}

message Person
{
    required string name = 1;
    required int32 id = 2;
    optional string email = 3;
}

message FamilyMember {
    required int32 age = 1;
    required string name = 2;
}

message Family {
    repeated FamilyMember member = 1;
}

message BlobShape {
    repeated int64 dim = 1 [packed = true];
}

message BlobProto {
    optional BlobShape shape = 7;
    repeated float data = 5 [packed = true];
    repeated float diff = 6 [packed = true];
    repeated double double_data = 8 [packed = true];
    repeated double double_diff = 9 [packed = true];

    // 4D dimensions -- deprecated.  Use "shape" instead.
    optional int32 num = 1 [default = 0];
    optional int32 channels = 2 [default = 0];
    optional int32 height = 3 [default = 0];
    optional int32 width = 4 [default = 0];
}